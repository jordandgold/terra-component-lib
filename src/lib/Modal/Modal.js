import React from "react";
import Button from "../Button/Button";
import "./Modal.scss";

const Modal = props => {
  const { className, title, body, buttonOne, buttonTwo, closeModal } = props;
  return (
    <aside className={`modal ${className}`}>
      <button onClick={closeModal()}>close</button>
      <h4>{title}</h4>
      <p>{body}</p>
      {buttonOne && (
        <Button
          onClick={buttonOne.onClick}
          text={buttonOne.text}
          className={buttonOne.className}
        />
      )}
      {buttonTwo && (
        <Button
          onClick={buttonTwo.onClick}
          text={buttonTwo.text}
          className={buttonTwo.className}
        />
      )}
    </aside>
  );
};

export default Modal;
