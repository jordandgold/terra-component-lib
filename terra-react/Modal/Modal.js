import React from "react";
import Button from "../Button/Button";
import Icon from "../Icon/Icon";
import "./Modal.scss";

var Modal = function Modal(props) {
  var className = props.className,
      title = props.title,
      body = props.body,
      buttonOne = props.buttonOne,
      buttonTwo = props.buttonTwo,
      closeModal = props.closeModal,
      name = props.name;
  return React.createElement("aside", {
    className: "ter-modal ".concat(className)
  }, React.createElement("button", {
    className: "ter-modal__close",
    onClick: function onClick() {
      return closeModal(name);
    },
    "aria-label": "Close Modal"
  }, React.createElement(Icon, {
    type: "enclosed-x-dark-16px",
    size: "16px"
  })), React.createElement("div", {
    className: "ter-modal__body"
  }, React.createElement("h4", {
    className: "ter-modal__title"
  }, title), React.createElement("p", null, body), buttonOne && React.createElement(Button, {
    onClick: buttonOne.onClick,
    text: buttonOne.text,
    className: buttonOne.className
  }), buttonTwo && React.createElement(Button, {
    onClick: buttonTwo.onClick,
    text: buttonTwo.text,
    className: buttonTwo.className
  })));
};

export default Modal;