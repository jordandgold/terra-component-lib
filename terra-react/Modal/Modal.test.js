import React from "react";
import { shallow } from "enzyme";
import Modal from "./Modal";
describe("Modal", function () {
  var wrapper;
  var mockOnClickOne;
  var mockOnClickTwo;
  var mockCloseModal;
  var buttonOne;
  var buttonTwo;
  beforeEach(function () {
    mockOnClickOne = jest.fn();
    mockOnClickTwo = jest.fn();
    mockCloseModal = jest.fn();
    buttonOne = {
      onClick: mockOnClickOne,
      text: "button one mock text",
      className: "mock class",
      name: "hello"
    };
    buttonTwo = {
      onClick: mockOnClickTwo,
      text: "button one mock text",
      className: "mock class"
    };
    wrapper = shallow(React.createElement(Modal, {
      name: "hello",
      className: "test",
      title: "test title",
      body: "test body",
      buttonOne: buttonOne,
      buttonTwo: buttonTwo,
      closeModal: mockCloseModal
    }));
  });
  it("should match the snap shot", function () {
    expect(wrapper).toMatchSnapshot();
  });
  it("should match the snap shot with one button", function () {
    wrapper = shallow(React.createElement(Modal, {
      className: "test",
      title: "test title",
      body: "test body",
      buttonOne: buttonOne,
      closeModal: mockCloseModal
    }));
    expect(wrapper).toMatchSnapshot();
  });
  it("should match the snap shot with no buttons", function () {
    wrapper = shallow(React.createElement(Modal, {
      className: "test",
      title: "test title",
      body: "test body",
      closeModal: mockCloseModal
    }));
    expect(wrapper).toMatchSnapshot();
  });
  it("should call close modal on click", function () {
    wrapper.find(".ter-modal__close").simulate("click");
    expect(mockCloseModal).toHaveBeenCalledWith("hello");
  });
});