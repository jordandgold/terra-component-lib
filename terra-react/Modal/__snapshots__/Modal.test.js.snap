// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match the snap shot 1`] = `
<aside
  className="ter-modal test"
>
  <button
    aria-label="Close Modal"
    className="ter-modal__close"
    onClick={[Function]}
  >
    <Icon
      size="16px"
      type="enclosed-x-dark-16px"
    />
  </button>
  <div
    className="ter-modal__body"
  >
    <h4
      className="ter-modal__title"
    >
      test title
    </h4>
    <p>
      test body
    </p>
    <Button
      className="mock class"
      onClick={[MockFunction]}
      text="button one mock text"
    />
    <Button
      className="mock class"
      onClick={[MockFunction]}
      text="button one mock text"
    />
  </div>
</aside>
`;

exports[`Modal should match the snap shot with no buttons 1`] = `
<aside
  className="ter-modal test"
>
  <button
    aria-label="Close Modal"
    className="ter-modal__close"
    onClick={[Function]}
  >
    <Icon
      size="16px"
      type="enclosed-x-dark-16px"
    />
  </button>
  <div
    className="ter-modal__body"
  >
    <h4
      className="ter-modal__title"
    >
      test title
    </h4>
    <p>
      test body
    </p>
  </div>
</aside>
`;

exports[`Modal should match the snap shot with one button 1`] = `
<aside
  className="ter-modal test"
>
  <button
    aria-label="Close Modal"
    className="ter-modal__close"
    onClick={[Function]}
  >
    <Icon
      size="16px"
      type="enclosed-x-dark-16px"
    />
  </button>
  <div
    className="ter-modal__body"
  >
    <h4
      className="ter-modal__title"
    >
      test title
    </h4>
    <p>
      test body
    </p>
    <Button
      className="mock class"
      onClick={[MockFunction]}
      text="button one mock text"
    />
  </div>
</aside>
`;
